apps/showthis/showthis.star

```diff
@@ -0,0 +1,142 @@
+
+def main(config):
+
+	url = config.get("url")
+	# url = "test" # query a test url
+
+
+	if url == "test":
+		url = "https://hook.integromat.com/ujwv9g2ug7budr8stcb5tvn9bjtrrb5m"
+
+	if url == "" or url == None:
+		print("Error: No URL configured")
+
+		display_vals = {}
+		display_vals["text_left"] = "ShowThis"
+		display_vals["text_right"] = "Err"
+		display_vals["text_large"] = "Please"
+		display_vals["text_small"] = "configure URL"
+		display_vals["icon"] = DEFAULT_ICON
+
+	else:
+		display_vals_json = cache.get("showthis_url " + url) 
+
+		if  display_vals_json != None:
+			print("Cache hit! Displaying cached data.")
+			
+			display_vals = json.decode(display_vals_json)
+
+		else:
+			print("No cache => Querying web service...")
+			
+			rep = http.get(url)
+		
+			if rep.status_code == 200:
+				display_vals = rep.json()
+
+				if display_vals != None:
+
+					cache_ttl_sec = CACHE_TTL_MINUTES * 60
+					cache.set("showthis_url " + url, json.encode(display_vals), ttl_seconds=cache_ttl_sec)
+
+				else:
+					display_vals["text_large"] = "Error"
+					display_vals["text_small"] = "Invalid obj"
+					display_vals["text_left"] = "ShowThis"
+					display_vals["text_right"] = "Err"
+					display_vals["icon"] = DEFAULT_ICON
+
+			else:
+				fail("Service request failed with status %d", rep.status_code)
```

Comment:
Let's drop this `fail`. It will kill the execution of the app immediately, so the error code will never be display to the user below.

Reply:
Thanks @rohansingh! Statement was meant for testing purposes, I forgot to remove it.